src/sidebar/services/stream-filter.js

```diff
@@ -1,3 +1,46 @@
+/**
+ * @typedef {'equals'|'one_of'} Operator
+ */
+
+/**
+ * Filter clause against which annotation updates are tested before being
+ * sent to the client.
+ *
+ * @typedef FilterClause
+ * @prop {string} field
+ * @prop {Operator} operator
+ * @prop {any} value
+ * @prop {boolean} case_sensitive - TODO: Backend doesn't use this at present,
+ *   but it seems important for certain fields (eg. ID).
+ */
+
+/**
+ * @typedef Filter
+ * @prop {string} match_policy - TODO: Remove this, the backend doesn't use it any more.
+ * @prop {FilterClause[]} clauses
+ * @prop {Object} actions - TODO: Remove this, the backend doesn't use it any more.
```

Comment:
This is a bit confusing to me. Maybe I just need to understand TS a bit more. But its not clear when we can get away with {Object} vs something custom ...

```
@typedef Actions
@prop {boolean} [create]
@prop {boolean} [update]
@prop {boolean} [delete]
```

Reply:
There are two different uses of `Object` to be aware of:

1. The first is to define a property or parameter type as "some kind of value that is not null/undefined" but you don't have enough information to be more specific. `Function` is similar, but for functions specifically.
2. The second use case, which is what we're doing here, is defining an object type "inline" as part of a parameter definition. The first line has `@param {Object} <parameter name>` and that is followed by several `@param {<property type>} <parameter name>.<property name>` lines.